features/step_definitions/assertion_steps.rb

```diff
@@ -1,150 +1,95 @@
+Then(/^I should( not)? see #{CAPTURE_STRING} image$/) do |negate, alt_text|
+  expect(page).send (negate ? :not_to : :to),  have_xpath("//img[contains(@alt,'#{alt_text}')]")
 end
 
-And(/^I should not see button t\("([^"]*)"\)$/) do |button|
-  expect(page).not_to have_button i18n_content(button)
-end
 
-And(/^I should see the checkbox with id "([^"]*)" unchecked$/) do |checkbox_id|
-  expect(page).to have_unchecked_field checkbox_id
+Then(/^I should( not)? see button #{CAPTURE_STRING}$/) do |negate, button|
+  expect(page).send (negate ? :not_to : :to),  have_button(button)
 end
 
-And(/^I should not see the checkbox with id "([^"]*)" unchecked$/) do |checkbox_id|
-  expect(page).not_to have_unchecked_field checkbox_id
-end
 
-And(/^I should see the checkbox with id "([^"]*)" checked/) do |checkbox_id|
-  expect(page).to have_unchecked_field checkbox_id
+Then(/^I should( not)? see #{CAPTURE_STRING} link$/) do |negate, link_label|
+  expect(page).send (negate ? :not_to : :to),  have_link(link_label)
 end
 
-And(/^I should not see the checkbox with id "([^"]*)" checked/) do |checkbox_id|
-  expect(page).not_to have_unchecked_field checkbox_id
-end
 
-And(/^I should see "([^"]*)" link$/) do |link_label|
-  expect(page).to have_link link_label
+Then(/^I should( not)? see the checkbox with id "([^"]*)" unchecked$/) do |negate, checkbox_id|
+  expect(page).send (negate ? :not_to : :to),  have_unchecked_field(checkbox_id)
 end
 
-And(/^I should see t\("([^"]*)"\) link$/) do |link_label|
-  expect(page).to have_link i18n_content(link_label)
-end
 
-And(/^I should not see "([^"]*)" link$/) do |link_label|
-  expect(page).not_to have_link link_label
-end
-
-And(/^I should not see t\("([^"]*)"\) link$/) do |link_label|
-  expect(page).not_to have_link i18n_content(link_label)
-end
-
-
-Then(/^I should be on (?:the )*"([^"]*)" page$/) do |page|
-  case page.downcase
-    when  'login'
-      path = new_user_session_path
-    when 'landing'
-      path = root_path
-    when 'edit my application'
-      path = edit_membership_application_path(@user.membership_applications.last)
-    when 'show my application'
-      path = membership_application_path(@user.membership_applications.last)
-    when 'user instructions'
-      path = information_path
-    when 'member instructions'
-      path = information_path
-    when 'all waiting for info reasons'
-      path = admin_only_member_app_waiting_reasons_path
-    when 'new waiting for info reason'
-      path = new_admin_only_member_app_waiting_reason_path
-    when 'new waiting for info reason'
-      path = new_admin_only_member_app_waiting_reason_path
-    when 'register as a new user'
-      path = new_user_registration_path
-    when 'edit registration for a user'
-      path = edit_user_registration_path
-  end
-
-  expect(current_path_without_locale(current_path)).to eq path
+Then(/^I should be on (?:the )*"([^"]*)" page(?: for "([^"]*)")?$/) do |page, email|
```

Comment:
Wow ... I didn't know you could have a capture group inside of a non-capture group - pretty cool.

Reply:
Hound complaining about the _Useless assignment to variable - path_ was because I had forgotten to return the path after the case statement. But I already fixed that in my latest commit.